### DISCUSSION TITLE
I’d like to propose an enhanced team structure for our "ml-frontend" project

### DISCUSSION THREAD

**Discussion by jinmayamashita** [monstar-lab-oss/ml-frontend#1220] [Ideas]
Hey ML FE Team,

We're considering a shift towards a three-team structure—Best Practice, CLI, and Modules—with the goal of refining our development process to better leverage our strengths and foster deeper collaboration. Here's a brief overview of how this might look:

![Untitled-2024-03-22-1851](https://github.com/monstar-lab-oss/ml-frontend/assets/9401060/ebc17404-b848-484f-a922-a559b961ca45)

## Best Practice Team
Our foundation. This team focuses on setting and maintaining high coding standards to ensure our project remains scalable, maintainable, and above all, consistent.

- Base Code: Establishing robust coding practices that everyone adheres to, ensuring our base code is clean and efficient.
- Linting: Implementing and enforcing linting rules that keep our code in check, preventing common errors and maintaining stylistic consistency.

## CLI Team
Our efficiency experts. This team enhances our development workflow by providing tools and scripts that automate repetitive tasks and streamline our processes.

- Plop & Codegen: We'll be building systems that automatically generate code for us, reducing redundancy and errors, and letting us focus on innovation.
- Helper Functions: Developing a library of reusable code snippets and functions that solve common problems, making our codebase more efficient and interconnected.
- Prompt & Docs: Overseeing command-line interfaces and prompts to ensure they are intuitive and guide developers smoothly through various tasks.

## Modules Team

The heart of our project. This team is responsible for developing and testing the individual components that come together to form our applications.

- Implementation: Building out the functional aspects of our web applications, ensuring each module not only works well but integrates seamlessly with the whole.
- Testing: Embracing a rigorous testing regime, including unit tests, UI component tests with Storybook, and E2E tests to guarantee reliability and a flawless user experience.

**Comment by ptrkdan** [comment]
I think that this is an excellent idea - each team would be more focused and (hopefully) efficient. 
One concern I have is, since we don't have fixed team members due to higher priority projects, it may be difficult to properly split the work into teams. 
I would like to hear more about this on our next meeting, though!

**Reply by ptrkdan** [reply]
03/29 Meeting Notes 
Participants: @jinmayamashita @seiya0914 @ptrkdan
- At the very least, we can assign one core member to "lead" each group.
  - Core member will plan and organize the tasks, and communicate with other groups
- Since tasks for each group will not always be evenly distributed, we can reallocate members as necessary. 
  - i.e. currently, the highest priority is improving the CLI, so we can allocate more members to CLI tasks.

**Comment by jinmayamashita** [comment]
I've labeled the recent issues for each team in Issues/PRs/Discussions! Regarding the issues, it seems appropriate for each team to manage or the lead to set whether they want to resolve them and by when.

- Best practice: [PRs](https://github.com/monstar-lab-oss/ml-frontend/pulls?q=is:pr+is:open+label:%22Best+practice%22) 
- Modules: [Discussions](https://github.com/monstar-lab-oss/ml-frontend/discussions?discussions_q=is:open+label:Modules) [Issues](https://github.com/monstar-lab-oss/ml-frontend/issues?q=is:issue+is:open+label:Modules) 
- CLI: [Discussions](https://github.com/monstar-lab-oss/ml-frontend/discussions?discussions_q=is:open+label:CLI) [Issues](https://github.com/monstar-lab-oss/ml-frontend/issues?q=is:issue+is:open+label:CLI)

**Comment by ptrkdan** [comment]
Current team structure:
- CLI
  - @jinmayamashita 
  - @G-N555 
- Modules
  - @ptrkdan 
- Best Practice
  - @mvn-luanvu-hn 
  - @seiya0914
